pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/groupby/DefaultGroupKeyGenerator.java

```diff
@@ -64,12 +66,19 @@
     ARRAY_MAP_BASED
   }
 
+  // Enum to reflect if trimming of group keys is ON or OFF
+  private enum TrimMode {
```

Comment:
do we need this? we already have the ability to set maxCapacity rt?can we simply set it to very high value to turn off the trim mode